plugins/outputs/nats/nats_test.go

```diff
@@ -263,3 +266,230 @@ func createStream(t *testing.T, server string, cfg *nats.StreamConfig) {
 	_, err = js.AddStream(cfg)
 	require.NoError(t, err)
 }
+func TestWriteWithLayoutIntegration(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration test in short mode")
+	}
+	natsServicePort := "4222"
+	type testConfig struct {
+		name                    string
+		container               testutil.Container
+		nats                    *NATS
+		streamConfigCompareFunc func(*testing.T, *jetstream.StreamInfo)
+		wantErr                 bool
```

Comment:
Please split the test into one instance that expects an error and one where you _don't_ expect an error. Sort the cases accordingly. This provides some context for the reader and simplifies the test code.